### DISCUSSION TITLE
[Deprecated] pinia-plugin-on-mounted

### DISCUSSION THREAD

**Discussion by Gumball12** [vuejs/pinia#2224] [Show and tell]
> Hi everyone. I just deployed a plugin that runs a callback when Pinia is mounted.
> Usage is simple:
> ```ts
> import { onPiniaMounted } from 'pinia-plugin-on-mounted';
> 
> onPiniaMounted(() => {
>   // do something
> });
> ```
> https://github.com/Gumball12/pinia-plugin-on-mounted
> Thanks!

The `onPiniaMounted()` hook doesn't seem to be necessary. See comments!

**Comment by posva** [comment]
It’s great you created a plugin but this kind of hook isn’t needed to call useStore within component setup like shown in the Readme  😅

**Reply by Gumball12** [reply]
Thanks for the feedback!

As you said, you are correct that `useStore()` should be used inside the component setup. However, there were some cases where I needed the store outside of the component and wanted to call `useStore()` before Pinia was initialized. This hook was created for those situations.

```ts
import { useUserStore } from '@/stores/user';
import { createPinia } from 'pinia';
import { createApp } from 'vue';
import App from './App.vue';

// fails because it's called before the Pinia is created
// const userStore = useUserStore();

// I know this wasn't the best idea, but...
onPiniaMounted(() => {
  const userStore = useUserStore();
});

const pinia = createPinia();
const app = createApp(App);
app.use(pinia);

// This is the way it's supposed to be
const userStore = useUserStore();
```

However, it's also possible that I've completely misunderstood and used Pinia incorrectly, and if so... I'll use that way instead of using this hook.

Thanks for making awesome open source! 🥰

**Reply by posva** [reply]
I see, in that scenario, passing the pinia instance is enough

**Reply by Gumball12** [reply]
You're right, I think I may have misunderstood something.
Creating a Pinia instance and then working with it is correct, anything else seems to be the wrong way to go.
Thanks for the quick help, I think I'm on the right track ☺️
